src/components/PublicHeader.tsx

```diff
@@ -10,15 +10,17 @@ import {
 } from '@adobe/react-spectrum';
 
 import apiUrl from '../utils/apiUrl';
-import { ZetkinOrganization, ZetkinUser } from '../types/zetkin';
+import { ZetkinMembership, ZetkinOrganization, ZetkinUser } from '../types/zetkin';
 
 interface PublicHeaderProps {
     user: ZetkinUser | null;
     org?: ZetkinOrganization | null;
     canOrganize?: boolean;
+    officials: ZetkinMembership[];
```

Comment:
The public interface (i.e. the props) that this component exposes to the wold should be as simple for component users (programmers) to understand as possible, and difficult to misunderstand.

In my opinion, there are two flaws with the way it currently looks.

## 1. The name and content of `officials`
The name of the new property `officials` implies that what should be passed is a list of organization officials, which could mean a list of several people who are all officials in an organization (or several). What's actually being passed is a list of memberships belonging to a single user. I think the name used for the variable in `DefaultLayout` is more clear, and should be used as a name for the property as well, i.e. `officialMemberships`.

Optionally, the property could be repurposed to pass in _all user memberships_, which would then be filtered inside the `PublicHeader` component (currently happens in `DefaultLayout`). With that solution, less would need to happen in `DefaultLayout` (which is difficult to test because it's [not pure](https://medium.com/technofunnel/working-with-react-pure-components-166ded26ae48)) and more could be done inside `PublicHeader`, which is easier to test (because it _is_ pure).

## 2. Derived data property
There are now two properties that both contain information about whether the user should see the _Organize_ button or not, `officials` and `canOrganize`. Actually, one is actually supposed to be calculated from the other, i.e. `canOrganize` depends on what the `officials` array contains, it is said to be _derived from_ `officials`, which is _base data_.

The issue is that both the _derived data_ and _base data_ needs to be calculated by the component that _uses_ `PublicHeader`. The `PublicHeader` component is basically telling anyone who wants to use it that it must provide two properties, one of which could be calculated from the other, but does not take responsibility of how that calculation should happen.

What happens if they contain conflicting information, or one is omitted? If the developer that uses `PublicHeader` somewhere forgets to include the `canOrganize` property (which is declared optional above) but passes in `officials`. Or if they set `canOrganize` to `true` but pass in an empty array of `officials`? At least with the way it's implemented right now, the latter case would result in an exception.

I would remove `canOrganize` now that we have `officials` (albeit renamed as suggested in 1 above).